api/v1beta1/clusterclass_types.go

```diff
@@ -415,6 +420,7 @@ type MachinePoolClass struct {
 	// Must match a key in the FailureDomains map stored on the cluster object.
 	// NOTE: This value can be overridden while defining a Cluster.Topology using this MachinePoolClass.
 	// +optional
+	// +kubebuilder:validation:MaxItems=1000
```

Comment:
This seems kinda high from a cloud user perspective, I guess in a metal datacenter someone may have many many failure domains if they consider each rack as one domain maybe?

Reply:
Yeah. I'm not sure. I was first considering 100. But then I don't know what weird things folks might be doing. I'm fine with 200-500 as well

Reply:
I'll go with 100. If folks complain we can increase.